app/allocation/views/view.njk

```diff
@@ -31,111 +40,100 @@
-    <section class="govuk-!-margin-top-8">
-    <h2 class="govuk-heading-m govuk-!-margin-bottom-1">{{ t("allocations::view.people_assign.heading") }}</h2>
-    {% if allocationPeople.emptySlots %}
-    <p>{{ t("allocations::view.people_assign.special_vehicle_instruction", {href:"/move/new"}) | safe }}</p>
-
-    <h3 class="govuk-heading-s">{{ t("allocations::view.people_assign.slots_empty", {
-      slots_empty: allocationPeople.emptySlots,
-      person_or_people: t('person', {
-        count: allocationPeople.emptySlots
-      })
-    }) }}</h3>
-      <p>{{ govukButton({
-        html: "Add person",
-        href: "/move/" + unassignedMoveId + "/assign"
-      }) }}</p>
-    {% endif %}
-    {% if allocationPeople.filledSlots.length === 0 %}
-      {{ emptySlot(t("allocations::view.people_assign.no_slots_filled")) }}
+    <h3 class="govuk-heading-s">
+      {{ t("allocation::view.people.progress", {
+        count: remainingCount if canAccess('allocation:person:assign') else totalCount,
+        addedCount: addedCount,
+        context: "remaining" if canAccess('allocation:person:assign')
+      }) }}
+    </h3>
+
+    {% if canAccess('allocation:person:assign') and remainingCount %}
+      <p>
+        {{ t("allocation::view.people.special_vehicle_instruction", {
+          href:"/move/new"
+        }) | safe }}
+      </p>
+
+      {{ govukButton({
+        href: "/move/" + unassignedMoveId + "/assign",
+        html: t("actions::add_person", {
+          context: "another" if addedCount !== 0
+        })
+      }) }}
     {% endif %}
   </section>
-  {% else %}
-    <section class="govuk-!-margin-top-8">
-    <h2 class="govuk-heading-m govuk-!-margin-bottom-1">{{ t("allocations::view.people.heading") }}</h2>
-    <h3 class="govuk-heading-s">{{ t("allocations::view.people.slots_filled", {
-      slots_filled: allocationPeople.filledSlots.length,
-      total_slots: allocation.moves_count,
-      person_or_people: t('person', {
-        count: allocation.moves_count
-      })
-    }) }}</h3>
-  </section>
 
-  {% for i in range(0, allocationPeople.emptySlots) -%}
-      {{ emptySlot(t("allocations::view.people.empty_slot")) }}
-    {%- endfor %}
-  {% endif %}
+  {% for move in moves %}
+    <div class="app-!-position-relative">
+      {{ appCard(move.card) }}
 
-  {% for slot in allocationPeople.filledSlots %}
-    <section class="app-!-position-relative">
-      {{ appCard(slot.card) }}
       {# TODO: consider handling actions within card component #}
       {# TODO: consider whether long names may cause overlap problems #}
-      {% if canAccess('allocation:person:assign') %}
+      {% if move.person and canAccess('allocation:person:assign') %}
         <p class="app-!-position-top-right govuk-!-margin-top-3">
           {{ govukButton({
-            href: '/move/' + slot.id + "/unassign",
-            html: t("allocations::view.person_unassign", {name: slot.fullname}),
+            href: '/move/' + move.id + "/unassign",
```

Comment:
As above, maybe use template strings.

Reply:
Same as above. I'm not aware it's possible to use template strings in a nunjucks file.

Although this has highlighted the mix of single/double quotes which I'll fix...

Reply:
++1 to being consistent with our quotes in the nunjucks templates.

Should we consider linting them?

But at a deeper level, I think we should be computing these values outside of the templates if we can (especially urls). As little logic and assignment of values in the templates as possible.

Reply:
I'd be keen to lint, but I wasn't aware there was a way to easily lint nunjucks files?

So up until now it's unfortunately been a consistency I've just tried to preserve via code reviews.

I do agree it would also be good to tidy up some of the large view files. They're containing quite a lot of logic that isn't easily as easy to test with just end-to-end tests.